pkg/cloudprovider/aws/instancetype.go

```diff
@@ -135,9 +138,18 @@ func (i *InstanceType) memory() resource.Quantity {
 	)
 }
 
-// Setting ephemeral-storage to be arbitrarily large so it will be ignored during binpacking
+// Setting ephemeral-storage to be either the default value or what is defined in blockDeviceMappings
 func (i *InstanceType) ephemeralStorage() resource.Quantity {
-	return resource.MustParse("100Pi")
+	EphemeralBlockDevice := amifamily.GetAMIFamily(i.provider.AMIFamily, &amifamily.Options{}).EphemeralBlockDevice()
+	if len(i.provider.BlockDeviceMappings) > 0 {
```

Comment:
Nit, but I don't think this check on the slice length is necessary, you can iterate over empty/nil slices.